app/javascript/components/game_map.js

```diff
@@ -0,0 +1,63 @@
+import { Axis } from './axis'
+
+class GameMap {
+  constructor(selector) {
+    this._selector = selector
+  }
+
+  addMap() {
+    // Add component html to DOM
+    this._element.innerHTML = this._html()
+
+    // Get the canvas context for the map layer and draw the map image
+    this._drawMapImage(
+      document.getElementById('map-component__map-layer').getContext('2d')
+    )
+
+    this._drawHexGrid(
+      document.getElementById('map-component__grid-layer').getContext('2d')
+    )
+  }
+
+  _axes() {
+    var q = new Axis(JSON.parse(this._element.dataset.q))
+    var r = new Axis(JSON.parse(this._element.dataset.r))
+    var s = new Axis(JSON.parse(this._element.dataset.s))
+
+    return { q, r, s }
+  }
+
+  _drawHexGrid(ctx) {
+    var axes = this._axes()
+    axes.q.drawLines(ctx)
+    axes.r.drawLines(ctx)
+    axes.s.drawLines(ctx)
```

Comment:
Instead of hardcoding the names of the axes, can this just draw all lines contained in each axis?